emukit/quadrature/loop/bq_stopping_conditions.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020-2024 The Emukit Authors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import logging
+
+import numpy as np
+
+from ...core.loop.stopping_conditions import StoppingCondition
+from .bq_loop_state import QuadratureLoopState
+
+_log = logging.getLogger(__name__)
+
+
+class CoefficientOfVariationStoppingCondition(StoppingCondition):
+    r"""Stops once the coefficient of variation (COV) falls below a threshold.
+
+    The COV is given by
+
+    .. math::
+        COV = \frac{\sigma}{\mu}
+
+    where :math:`\mu` and :math:`\sigma^2` are the current mean and variance respectively of the integral according to
+    the BQ posterior model.
+
+    :param eps: Threshold under which the COV must fall.
+    :param delay: Number of times the stopping condition needs to be true in a row in order to stop. Defaults to 1.
+
+    :raises ValueError: If `delay` is smaller than 1.
+    :raises ValueError: If `eps` is non-negative.
+
+    """
+
+    def __init__(self, eps: float, delay: int = 1) -> None:
+
+        if delay < 1:
```

Comment:
```suggestion
        if delay < 1 or not isinstance(delay, int):
```
because if you pass float('inf'), then you will have an infinite loop right?